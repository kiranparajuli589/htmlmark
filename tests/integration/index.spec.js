const { pathJoin } = require("../../lib/util/file")
const { toHtml } = require("../../lib/index")


describe("To HTML", () => {
  it.each([
    // eslint-disable-next-line no-undef
    pathJoin(__dirname, "..", "fixtures/markdown.md"),
    // eslint-disable-next-line no-undef
    pathJoin(__dirname, "..", "fixtures/large_markdown.md")
  ])("should parse the markdown file content to html", (path) => {
    const html = toHtml(path)
    expect(html).toMatchSnapshot()
  })
})
