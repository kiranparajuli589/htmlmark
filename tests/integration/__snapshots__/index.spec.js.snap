// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To HTML should parse the markdown file content to html 1`] = `
"
<br>
<h1>Heading One <code>CODE INSIDE</code></h1>
<h2>Heading Two</h2>
<h3>Heading Three</h3>
<h4>Heading Four</h4>
<h5>Heading Five</h5>
<h6>Heading Six</h6>
<br>
<p>a paragraph of words <code>first code</code> normal text here <code>code body</code> <em>first italics</em> here me crying <em>italic body</em> here me crying <strong>first bolds</strong> some normal again <strong>bold body</strong> <a href='https://kiranparajuli.com.np'>Kiran Parajuli</a> <s>strikes body</s> here some normal again at the last</p>
<br><blockquote>
<p> some quote with <u>Underline</u></p>
</blockquote>
<br>
<ul>
  <li>list item 1</li><li>list item 2</li>
</ul>
<ul>
  <li>list item 3</li><li>list item 4</li>
</ul>
<ul>
  <li>list item 5</li><li>list item 6</li>
</ul>
<ul>
  <li><input type=\\"checkbox\\">checkbox empty</li><li><input type=\\"checkbox\\" checked>checkbox checked</li>
</ul>
<ul>
  <li><input type=\\"checkbox\\">checkbox empty</li><li><input type=\\"checkbox\\" checked>checkbox checked</li>
</ul>
<ul>
  <li>checkbox empty</li><li>checkbox checked</li>
</ul>
<ol>
  <li>numbered list item 1</li><li>numbered list item 2</li>
</ol>
<ol>
  <li><input type=\\"checkbox\\">numbered checkbox empty</li><li><input type=\\"checkbox\\" checked>numbered checkbox checked</li>
</ol>
<p>paragraph just below hmm, but with some fun :)</p>
<br><blockquote>
<p> quote 1</p><blockquote>
<p> nested quote 2 <a href='quote-link-url'>quote-link</a></p>
</blockquote><blockquote><blockquote>
<p> nested quote 3 with <em>italics</em> and <strong>bolds</strong></p>
</blockquote></blockquote>
</blockquote>
<br>
<pre><code class='language-js'>const a = 1;
const b = 1;
const c = a + b;

if (c === 2) {
  console.log('success');
} else {
  console.log('failure');
}</code></pre>
<p>  some line with small space ahead</p>
<br>
<pre><code class='language-js'></code></pre>
<br>
<p>some piece of normal paragraph with <u>underlined text</u></p>
<br>
<pre><code>another code block is here</code></pre>
<br>
<img src=\\"https://avatars.githubusercontent.com/u/39373750?v=4\\" alt=\\"Kiran Parajuli\\">
<br>
<p>some random text</p>
<br>
<table>
  <thead>
    <tr><th>column 1</th><th>column 2</th><th>column 3</th></tr>
  </thead>
  <tbody>
    <tr><td>r1 text c1</td><td>r1 <u>text</u> c2</td><td>r1 text c3</td></tr><tr><td>r2 text c1</td><td>r2 <u>text</u> c2</td><td>r2 text c3</td></tr><tr><td>r3 text c1</td><td>r3 <strong>text</strong> c2</td><td>r3 text c3</td></tr><tr><td>r4 text c1</td><td>r4 <em>text</em> c2</td><td>r5 text c3</td></tr><tr><td>r6 text c1</td><td>r6 <s>text</s> c2</td><td>r6 text c3</td></tr>
  </tbody>
</table>
<p>some random text again just below table | abc | def | |---|---|</p>
<br>
<p>text text text <strong>bold <em>with</em> <code>gems</code> <a href='treasure-chest'>ruby</a> <s>please</s></strong> text text text text text text <em>italics <code>with</code> <a href='ruby'>gems</a> <s>please</s></em> text text text text text text <s>strikes <em>with</em> <code>gems</code> <strong>ruby</strong> <a href='no-sparkle'>gold</a></s> text text text text <u>text <strong>text</strong> <code>text</code> <em>text</em> <a href='xt'>te</a> text</u> text</p>
<br>
<p><a href='ruby'><em>bold</em> and <code>gem</code>ss</a> and some text</p>
<br>"
`;
